src/Engine/GameEvents/TeleportEvent.java

```diff
@@ -76,39 +81,39 @@ public void performEvent(MapMaster gameMap)
     gameMap.removeUnit(unit); // First remove our guy. He's in the ether.
 
     // Figure out if something's in the way, and what to do with it.
-    Unit oldResident = gameMap.getLocation(unitDestination).getResident();
-    boolean killOldResident = false;
-    if( null != oldResident )
+    obstacle = gameMap.getLocation(unitDestination).getResident();
+    obstacleDies = false;
+    if( null != obstacle )
     {
       switch(collisionOutcome)
       {
         case KILL:
-          killOldResident = true;
+          obstacleDies = true;
```

Comment:
Just a note: this should end up killing a unit that teleports onto its current tile. IDK if we care about that case, but it exists.

Reply:
Line 85 should prevent that.